### ISSUE TITLE
Ubisoft Connect Opens To A Black Screen / Crashes

### ISSUE THREAD

**Issue by intelligentgaming** [lutris/lutris#5412]
When opening Ubisoft Connect, you get a long black screen after the sign in process (as below).

![image](https://github.com/lutris/lutris/assets/67860954/6f69ff05-1b00-42ea-9f7d-75a30a4a00b4)

Replicated with the native Arch package and with Lutris Flatpak. 

This does eventually load after several minutes but is unusable. 

More than likely this is related to the Ubisoft Connect "New" style that has been rolled out. 

![image](https://github.com/lutris/lutris/assets/67860954/14739e7b-464a-4705-922b-d3772722f8f4)

The system is all updated as per this message.

**Comment by al12gamer** [comment]
What proton or wine are you using to open this app? I haven't used Ubisoft Connect in ages, but last I checked, it opened fine using the latest protonge

**Comment by intelligentgaming** [comment]
wine-ge-8.26-x86_64

**Comment by CountMurphy** [comment]
Same issue here, but I have a few more details that might help. On my intel graphics laptop (running wayland), ubisoft connect works flawlessly. On my Nvidia based gaming PC I too get the black screen. If I let the black screen sit for like 5 or ten minutes, whatever was causing the deadlock stops and I can use the client.

**Comment by CountMurphy** [comment]
Found a better work-around. Ubisoft connect works perfectly when running under wayland.

**Comment by AinoSpring** [comment]
I'm having a similar issue; my linux just gets shut down when I'm trying to run Ubisoft connect

**Comment by AinoSpring** [comment]
Yeah but the problem is that wayland for games is... kind of not working. Especially when running on Nvidia, which most people are, performance just doesn't exist.
I personally would be very interested on how you managed to run games performantly on wayland.

**Comment by intelligentgaming** [comment]
Hmm, strange, so I recently installed Ubuntu 23.10 and it is working.

![image](https://github.com/lutris/lutris/assets/67860954/d2886e86-a40d-45b2-a157-94270387767f)

So now I wonder if it was an Arch thing, so I guess this can be closed.

**Comment by CountMurphy** [comment]
With most distros still using X11 (steam deck included) and the massive problems with wayland on nvidia cards (screen tearing and graphical glitches), only supporting wayland is a very poor position to take.

**Comment by CountMurphy** [comment]
Odd, also running Arch

**Comment by intelligentgaming** [comment]
> Odd, also running Arch

Plasma or Gnome?

**Comment by CountMurphy** [comment]
Plasma and xfce4

**Comment by strycore** [comment]
Even if I were still running X11, I dont have the skills needed to solve the black screen on Ubisoft connect issue. I wouldn't even know where to begin.

I've used Wayland with Nvidia on a modern distro and noticed no issue (except for gamescope)

**Comment by st3iny** [comment]
Also running KDE on X11 with a nvidia card and it works again when using the `wine-ge-lol-8-27-x86_64` wine version.

**Comment by strycore** [comment]
Do not use ge-lol unless you are playing League of Legends. If something works with ge-lol but not ge, then something has gone very wrong

**Comment by st3iny** [comment]
Yeah, but it works. One thing I noticed is that Ubisoft Connect downgraded itself after switching to ge-lol. So maybe that might be the reason for why it works now.

**Comment by strycore** [comment]
Ubisoft Connect can downgrade itself??

**Comment by atahabaki** [comment]
I've faced with this issue recently. My machine basically a bumblebee setup, 2 graphics card on a laptop.

|X|Y|
|---|---|
|Distro|Arch|
|WM|Hyprland|
|iGPU|AMD Ryzen 7 5800HS w/ Radeon Graphics|
|dGPU| Nvidia RTX 3050 Mobile |

First, I launched Ubisoft Connect with dGPU... Faced with a giant blank screen. Exactly like the OP's screenshot. Then I thought, what about the iGPU. Switched to iGPU, then tried again. Voilla... It worked.

At a later time, I could debug, analyze it but for now... I will just use this workaround. Seems like it's related to Nvidia. Gonna look into it, later.

**Comment by josue2363** [comment]
I also have this same problem with Ubinsoft Connect when using Wine-GE-Proton8-26 with NVIDIA graphics card. Ubinsoft Connect works correctly with AMD Ryzen graphics.

**Comment by SlowNicoFish** [comment]
Can confirm this on Nvidia 4070 Super
Proton Experimental Bleeding edge and Wine-GE-proton8-26
OS: Arch Linux x86_64
Kernel: Linux 6.9.1-arch1-2
DE: Gnome 46.1
WM: Mutter (X11)
CPU: AMD Ryzen 7 5800X (16) @ 3.80 GHz
GPU: NVIDIA GeForce RTX 4070 SUPER [Discrete]

**Comment by st3iny** [comment]
So I switched to Wayland recently and it indeed works again on nvidia. With explicit sync changes being included in the current nvidia beta driver and KDE 6.1 beta, there is no reason to use/support X11 anymore. Even gaming works flawlessly now.

**Comment by josue2363** [comment]
This seems to help with the issue of Ubisoft Connect opening with a black window:
`export WINEDLLOVERRIDES="libglesv2.dll=d"`

**Comment by kxxt** [comment]
> Nice! Can anyone else confirms overriding this DLL solves the black screen issue on Nvidia / X11?

Yes. I could confirm it.

**Comment by strycore** [comment]
Cool. Now we can consider pushing that as the default, if it doesn't affect AMD and Intel users.

**Comment by josue2363** [comment]
I can confirm this does not affect AMD GPUs. I also have a Ryzen 3 2200g (Radeon Vega 8) and I can use Ubisoft Connect normally with this Wine environment variable.

**Comment by crm2704** [comment]
> Found a better work-around. Ubisoft connect works perfectly when running under wayland.

This was the answer for me, thanks

**Comment by SheMelody** [comment]
> This seems to help with the issue of Ubisoft Connect opening with a black window: `export WINEDLLOVERRIDES="libglesv2.dll=d"`

I can confirm that this helps, even when using Proton and Wine from commandline.

**Comment by SheMelody** [comment]
> Even if I were still running X11, I dont have the skills needed to solve the black screen on Ubisoft connect issue. I wouldn't even know where to begin.
> 
> I've used Wayland with Nvidia on a modern distro and noticed no issue (except for gamescope)

Wayland has problems with framepacing, especially during frontbuffer captures, on NVIDIA, AMD and Intel, it's just a Wayland thing. While these issues are not evident for some people, they're still there, and they make X11 a more viable option for us (at the moment).

**Comment by jasonsummers** [comment]
> This seems to help with the issue of Ubisoft Connect opening with a black window:
> 
> `export WINEDLLOVERRIDES="libglesv2.dll=d"`
> 
> 

This also helped me get past the black screen / hard crash issue.
Thank you very much!
